langwatch/src/pages/settings/roles.tsx

```diff
@@ -0,0 +1,897 @@
+import {
+  Badge,
+  Box,
+  Button,
+  Card,
+  Field,
+  Fieldset,
+  Heading,
+  HStack,
+  Input,
+  Separator,
+  Spinner,
+  Text,
+  Textarea,
+  VStack,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { Edit, Eye, Info, Plus, Shield, Trash2, Users } from "react-feather";
+import { ShieldUser } from "lucide-react";
+
+import { useForm } from "react-hook-form";
+import { useEffect, useState } from "react";
+import { Checkbox } from "../../components/ui/checkbox";
+import { Dialog } from "../../components/ui/dialog";
+import { Tooltip } from "../../components/ui/tooltip";
+import { toaster } from "../../components/ui/toaster";
+import SettingsLayout from "../../components/SettingsLayout";
+import { useOrganizationTeamProject } from "../../hooks/useOrganizationTeamProject";
+import { api } from "../../utils/api";
+import type { Permission, Resource, Action } from "../../server/api/rbac";
+import {
+  orderedResources,
+  getValidActionsForResource,
+} from "../../utils/permissionsConfig";
+
+import { withPermissionGuard } from "../../components/WithPermissionGuard";
+
+/**
+ * Role Management Settings Page
+ *
+ * Single Responsibility: Allows organization admins to create and manage custom roles
+ * with granular permission assignments.
+ */
+function RolesSettings() {
+  const { organization } = useOrganizationTeamProject();
+
+  if (!organization) {
+    return (
+      <SettingsLayout>
+        <VStack align="center" justify="center" width="full" height="200px">
+          <Spinner />
+        </VStack>
+      </SettingsLayout>
+    );
+  }
+
+  return (
+    <SettingsLayout>
+      <RolesManagement organizationId={organization.id} />
+    </SettingsLayout>
+  );
+}
+
+export default withPermissionGuard("organization:manage", {
+  layoutComponent: SettingsLayout,
+})(RolesSettings);
+
+type RoleFormData = {
+  name: string;
+  description: string;
+  permissions: Permission[];
+};
+
+function RolesManagement({ organizationId }: { organizationId: string }) {
```

Comment:
IMO, I think we should really try to keep the one component per file rule, unless we're doing a pure composible UI component that has the pattern `Component.Root`, `Component.Part`